training/tf/parse.py

```diff
@@ -270,21 +277,34 @@ def main(args):
     if not chunks:
         return
 
-    parser = ChunkParser(chunks)
+    # The following assumes positions from one game are not
+    # spread through chunks.
+    random.shuffle(chunks)
+    training, test = split_chunks(chunks, 0.1)
+    print("Training with {0} chunks, validating on {1} chunks".format(
+        len(training), len(test)))
 
-    run_test(parser)
+    #run_test(parser)
     #benchmark(parser)
 
+    train_parser = ChunkParser(training)
     dataset = tf.data.Dataset.from_generator(
-        parser.parse_chunk, output_types=(tf.string))
+        train_parser.parse_chunk, output_types=(tf.string))
     dataset = dataset.shuffle(1 << 18)
     dataset = dataset.map(_parse_function)
     dataset = dataset.batch(BATCH_SIZE)
     dataset = dataset.prefetch(4)
-    iterator = dataset.make_one_shot_iterator()
-    next_batch = iterator.get_next()
+    train_iterator = dataset.make_one_shot_iterator()
+
+    test_parser = ChunkParser(test, 1)
+    dataset = tf.data.Dataset.from_generator(
+        test_parser.parse_chunk, output_types=(tf.string))
+    dataset = dataset.map(_parse_function)
```

Comment:
Might be a good idea to shuffle the validation dataset too?

Reply:
Not sure, didn't we just discover the shuffle buffer may not help all that much unless its made much larger?